### PR TITLE
Gb/append econ

### PR DESCRIPTION
**Pull request by grantbuster**
Added an append option to the econ module to append data to the generation input h5 files. Should work great when needing to run gen -> econ as separate modules in the reV pipeline.

### CODE REVIEW

File: reV/config/sam_analysis_configs.py [hunk 1]
```diff
@@ -296,3 +297,29 @@ def site_data(self):
...
+
+        Returns
+        -------
+        dirout : str
+            Target path for reV output files.
+        """
+        self._dirout = super().dirout
+        if self.append:
+            self._dirout = os.path.dirname(self.cf_files[0])
+        return self._dirout
```
Comment (MRossol) [inline reV/config/sam_analysis_configs.py]: add line break

File: reV/econ/econ.py [hunk 2]
```diff
@@ -141,9 +144,13 @@ def __init__(self, points_control, cf_file, cf_year, site_data=None,
...
 
-        # initialize output file
-        self._init_fpath()
-        self._init_h5()
+        # initialize output file or append econ data to gen file
+        if append:
+            self._fpath = self._cf_file
+        else:
+            self._init_fpath()
+        mode = 'a' if append else 'w'
```
Comment (MRossol) [inline reV/econ/econ.py]: add line break

File: reV/econ/econ.py [hunk 3]
```diff
@@ -141,9 +144,13 @@ def __init__(self, points_control, cf_file, cf_year, site_data=None,
...
+        self._init_h5(mode=mode)
```
Comment (MRossol) [inline reV/econ/econ.py]: I don't understand how this doesn't break? Output.init_h5 will try to create all of the datasets which will break if they already exist. Am I missing something?

File: reV/handlers/outputs.py [hunk 4]
```diff
@@ -522,8 +522,13 @@ def _create_dset(self, ds_name, shape, dtype, chunks=None, attrs=None,
             Dataset attributes
         data : ndarray
             Dataset data array
+        replace : bool
+            If previous dataset exists with the same name, it will be replaced.
         """
         if self.writable:
+            if ds_name in self.datasets and replace:
+                del self.h5[ds_name]
+
```
Comment (MRossol) [inline reV/handlers/outputs.py]: need an elif that skips the create_dataset step.

Comment (grantbuster) [reply reV/handlers/outputs.py]: Previous behavior was to break on h5py "link already exists" error. I think this is important if the shape and dtype don't match. I'm adding a new commit that checks for this.

File: reV/econ/econ.py (hunk 3 above)
Comment (grantbuster) [reply reV/econ/econ.py]: Econ should only be making new datasets, plus I added better handling of pre-existing datasets.

**Comment by grantbuster** [comment]
@MRossol, approve? Travis used this last Friday and the results look good. I think its an important feature.

File: reV/handlers/outputs.py (hunk 4 above)
Comment (MRossol) [reply reV/handlers/outputs.py]: I'm surprised this logic works, the else won't be reached after you delete the dset so you won't re-create the dataset

Comment (grantbuster) [reply reV/handlers/outputs.py]: Yep you're right. fixed.

**Comment by grantbuster** [comment]
No problem! It's improved now. 

I dont think we need a new release? Its a new "feature" but super super minor.

**Comment by grantbuster** [comment]
@MRossol, actually, since Mike added in those solar thermal models I think it might be worth doing a feature release and might as well include this too.

**Pull request merged**